library-front/src/components/BookCard/BookCard.tsx

```diff
@@ -13,31 +17,76 @@ interface BookCardProps {
 
 const BookCard: FC<BookCardProps> = ({ book }) => {
   const { jwtToken } = useJwt()
-  const [bookModalVisible, setBookModalVisible] = useState(false)
+  const [visibleModal, setVisibleModal] = useState<string | undefined>()
+  const queryClient = useQueryClient()
 
   const role = jwtToken?.role
 
-  const showBookModal = () => {
-    setBookModalVisible(true)
+  const handleBookDelete = async () => {
+    const { data } = await getBook(book.Id).catch((err) => {
+      toast.error("Couldn't retrieve book")
+      throw new Error(err)
+    })
+    if (!data) return
+    if (data.Quantity !== data.Available) {
+      toast.warning("Book is currently being rented and can't be deleted")
+      return
+    }
+    setVisibleModal('delete')
+  }
+
+  const handleDeleteDecision = async (isConfirmed: boolean) => {
+    if (!isConfirmed) {
+      setVisibleModal(undefined)
```

Comment:
Stvarno? hahahaha

Reply:
To je posto imam state na osnovu koga se prikazuje modal. Ako je undefined onda se ne prikazuje tj sakriva se. Unutar modala imam content koji se prikazuje na osnovu te vrednosti. Ako je edit onda se prikazuje book form a ako je delete onda confirm modal